From ae159e63550d1a31af86638ad57b1015ff73b82e Mon Sep 17 00:00:00 2001
From: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Date: Wed, 12 Jan 2022 15:38:36 +0200
Subject: [PATCH 6/6] ddr: Update configuration for increasing r/w bandwidth

Updated the DDR Configuration registers to enable the
maximum read/write speed performance, as suggested by
the Validation team.

These changes fix the reported performance degradation issues
that occurred starting with BSP30.0.

Issue: ALB-7781
Upstream-Status: Pending 

Signed-off-by: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/nxp/s32g/ddr/s32g2/ddrc_cfg.c | 8 ++++----
 drivers/nxp/s32g/ddr/s32g3/ddrc_cfg.c | 8 ++++----
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/drivers/nxp/s32g/ddr/s32g2/ddrc_cfg.c b/drivers/nxp/s32g/ddr/s32g2/ddrc_cfg.c
index 920582d54..779c96497 100644
--- a/drivers/nxp/s32g/ddr/s32g2/ddrc_cfg.c
+++ b/drivers/nxp/s32g/ddr/s32g2/ddrc_cfg.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2020-2021 NXP
+ * Copyright 2020-2022 NXP
  *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions are met:
@@ -133,11 +133,11 @@ struct regconf ddrc_cfg[] = {
 	{0x403c022c, 0x00000000U},
 	{0x403c0240, 0x04000400U},
 	{0x403c0244, 0x00000000U},
-	{0x403c0250, 0x29001f05U},
+	{0x403c0250, 0x00001f01U},
 	{0x403c0254, 0x00000000U},
 	{0x403c025c, 0x10002200U},
-	{0x403c0264, 0x1f002100U},
-	{0x403c026c, 0x1f002100U},
+	{0x403c0264, 0x90000096U},
+	{0x403c026c, 0x1000012cU},
 	{0x403c0300, 0x00000000U},
 	{0x403c0304, 0x00000000U},
 	{0x403c030c, 0x00000000U},
diff --git a/drivers/nxp/s32g/ddr/s32g3/ddrc_cfg.c b/drivers/nxp/s32g/ddr/s32g3/ddrc_cfg.c
index 0df21618e..f556165e6 100644
--- a/drivers/nxp/s32g/ddr/s32g3/ddrc_cfg.c
+++ b/drivers/nxp/s32g/ddr/s32g3/ddrc_cfg.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2021 NXP
+ * Copyright 2021-2022 NXP
  *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions are met:
@@ -127,11 +127,11 @@ struct regconf ddrc_cfg[] = {
 	{0x403c022c, 0x00000000U},
 	{0x403c0240, 0x04000400U},
 	{0x403c0244, 0x00000000U},
-	{0x403c0250, 0x29001f05U},
+	{0x403c0250, 0x00001f01U},
 	{0x403c0254, 0x00000000U},
 	{0x403c025c, 0x10002200U},
-	{0x403c0264, 0x1f002100U},
-	{0x403c026c, 0x1f002100U},
+	{0x403c0264, 0x90000096U},
+	{0x403c026c, 0x1000012cU},
 	{0x403c0300, 0x00000000U},
 	{0x403c0304, 0x00000000U},
 	{0x403c030c, 0x00000000U},
-- 
2.17.1

